samples/objc/FirebaseUI-demo-objc/Samples/Auth/FUIAuthViewController.m

```diff
@@ -149,6 +151,29 @@ - (CGFloat)tableView:(UITableView *)tableView heightForRowAtIndexPath:(NSIndexPa
   return UITableViewAutomaticDimension;
 }
 
+- (void)tableView:(UITableView *)tableView didSelectRowAtIndexPath:(NSIndexPath *)indexPath {
+  if (indexPath.section == kSectionsAnonymousSignIn && indexPath.row == 0) {
+    if (_authUI.auth.currentUser.isAnonymous) {
+      [self showAlertWithTitlte:@"" message:@"Already signed in anonymously"];
+      [tableView deselectRowAtIndexPath:indexPath animated:NO];
+      return;
+    }
+    [self signOut];
+    [FUIAuth.defaultAuthUI.auth signInAnonymouslyWithCompletion:^(FIRUser *_Nullable user,
+                                                                  NSError *_Nullable error) {
+      if (error) {
+        NSError *detailedError = error.userInfo[NSUnderlyingErrorKey];
+        if (!detailedError) {
+          detailedError = error;
+        }
+        NSLog(@"ERROR: %@", detailedError.localizedDescription);
+        return;
```

Comment:
In case of error row is not unselected. Is that correct?

Reply:
Why would the row not be unselected.  If you are confused by the return statement it simply returns from the block but the code to unselect the row is certainly still executed. I have removed the return statement that seemed to confuse you.